File: tsp-solver-frontend/src/app/model/request/tsp/AntColonyRequest.ts
export class AntColonyRequest {
  costMatrix: number[][]
  alpha: number;
  beta: number;
  evaporationRate: number;
  q: number;
  antFactor: number;
  randomCitySelection: number;
  iterations: number;
  maxIterationsNoImprovement: number;

  constructor(costMatrix: number[][], alpha: number, beta: number, evaporationRate: number, q: number, antFactor: number,
              randomCitySelection: number, iterations: number, maxIterationsNoImprovement: number) {
    this.costMatrix = costMatrix;
    this.alpha = alpha;
    this.beta = beta;
    this.evaporationRate = evaporationRate;
    this.q = q;
    this.antFactor = antFactor;
    this.randomCitySelection = randomCitySelection;
    this.iterations = iterations;
    this.maxIterationsNoImprovement = maxIterationsNoImprovement;
  }

}

File: tsp-solver-backend/src/main/java/pl/edu/pbs/request/tsp/AntColonyRequest.java
package pl.edu.pbs.request.tsp;

import lombok.Data;

@Data
public class AntColonyRequest {

    private double[][] costMatrix;
    private double alpha;
    private double beta;
    private double evaporationRate;
    private double q;
    private double antFactor;
    private double randomCitySelection;
    private int iterations;
    private int maxIterationsNoImprovement;
}